cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/handler/async/RightsizeRequest.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2018 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.servlet.handler.async;
+
+import com.linkedin.kafka.cruisecontrol.servlet.handler.async.runnable.OperationFuture;
+import com.linkedin.kafka.cruisecontrol.servlet.handler.async.runnable.RightsizeRunnable;
+import com.linkedin.kafka.cruisecontrol.servlet.parameters.RightsizeParameters;
+import java.util.Map;
+
+import static com.linkedin.cruisecontrol.common.utils.Utils.validateNotNull;
+import static com.linkedin.kafka.cruisecontrol.servlet.parameters.ParameterUtils.RIGHTSIZE_PARAMETER_OBJECT_CONFIG;
+
+
+public class RightsizeRequest extends AbstractAsyncRequest {
```

Comment:
Should we make this request a sync request rather than an async request?

Reply:
Since we're using the `RightsizeEndpoint` to call the `Provisioner#rightsize` method, it would make sense for this to be an asynchronous process as it would be called in the `GoalViolationDetector`. There is no dependency in this case to have the request block the client until the operation completes.

Reply:
Since `Provisioner#rightsize` is a non-blocking method (i.e. based on its JavaDoc), it is not clear why calling it is relevant. All async requests in CC are the ones that require creation of a cluster model, which is a resource- (e.g. memory, cpu) and time-intensive (i.e. requires aggregation and synchronization) process. Hence, I feel that a sync request is a better fit here.